sw-src.js

```diff
@@ -55,99 +45,57 @@ self.registration.addEventListener("updatefound", function (e) {
-        //       )
-        //   : file
-//       )
-//       .finally(() => {
-//         number = number + 1;
-//         self.clients.matchAll().then((clients) => {
-//           clients.forEach((client) => {
-//             if ((number / (length * 5)) * 100 < 101) {
-//               client.postMessage({
-//                 msg: "Loading",
-//                 data: Math.round((number / (length * 5)) * 100),
-//               });
-//             } else {
-//               client.postMessage({
-//                 msg: "Loading",
-//                 data: Math.round(100),
-//               });
-//             }
-//           });
-//         });
-//       });
-//   });
-// }
+
 async function cacheLangAssets(file, cacheName) {
-  await caches.open(cacheName).then((cache) => {
-    cache.add(file);
-  });
+  const cache = await caches.open(cacheName);
+  const cachedResponse = await cache.match(file);
+
+  if (!cachedResponse) {
+    await cache.add(file);
+    console.log('Cached File:', file);
+  } else {
+    console.log('File already cached, skipping:', file);
+  }
 }
 async function getCacheName(language) {
-  ///awt
- await caches.keys().then((cacheNames) => {
+  await caches.keys().then((cacheNames) => {
     cacheNames.forEach(async (cacheName) => {
       await getALLAudioUrls(cacheName, language);
     });
   });
 }
 
 async function getALLAudioUrls(cacheName, language) {
-  // await cacheCommonAssets(language);
-  let audioList = [];
-  audioList.push("/lang/" + language + "/ftm_" + language + ".json");
+  let audioList = new Set(); // Use Set to filter duplicates
+  audioList.add("/lang/" + language + "/ftm_" + language + ".json");
+
   fetch("./lang/" + language + "/ftm_" + language + ".json", {
     method: "GET",
     headers: {
       "Content-Type": "application/json",
     },
   }).then((res) =>
     res.json().then(async (data) => {
-      await cacheFeedBackAudio(data.FeedbackAudios,language);
-      for (var i = 0; i < data.Levels.length; i++) {
-        
-
-        data.Levels[i].Puzzles.forEach(async (element) => {
-          let file = element.prompt.PromptAudio;
-          // audioList.push(element.prompt.PromptAudio)
-          audioList.push(
-          self.location.href.includes("https://feedthemonsterdev.curiouscontent.org")
-          ? file.slice(
-              0,
-              file.indexOf("/feedthemonster") + "/feedthemonster".length
-            ) +
-              "dev" +
-              file.slice(
-                file.indexOf("/feedthemonster") + "/feedthemonster".length
-              )
-          : file);
-        //  await cacheAudiosFiles(
-        //     element.prompt.PromptAudio,
-        //     workbox.core.cacheNames.precache + language,
-        //     data.Levels.length
-        //   );
-        });
+      await cacheFeedBackAudio(data.FeedbackAudios, language);
+
+      for (const level of data.Levels) {
+        for (const puzzle of level.Puzzles) {
```

Comment:
Question regarding the for loop, for the data.levels is this the 150 game levels that has 5 game puzzles?